workflow_scripts/src/ci.js

```diff
@@ -0,0 +1,148 @@
+        rootDir)
+        .then((code) =>{
+          if (code == 0) {
+            console.log('Install depot_tools: SUCCESS');
+          } else {
+            console.log('Install depot_tools: FAIL');
+            process.exit(1);
+          }
+        });
+  }
+
+  const separator = os.platform() == 'win32' ? ';' : ':';
+  process.env['PATH'] = depotToolsDir + separator + process.env['PATH'];
+
+  if (os.platform() == 'win32') {
+    process.env['DEPOT_TOOLS_WIN_TOOLCHAIN'] = '0';
+  }
+};
+
+const syncCode = async () => {
+  // gclient sync
+  fs.copyFileSync(path.join(rootDir, 'scripts', 'standalone.gclient'),
+      path.join(rootDir, '.gclient'));
+
+  process.chdir(rootDir);
+  await childProcessSpawn(
+    os.platform() == 'win32' ? 'gclient.bat' : 'gclient',
+    ['sync'], rootDir)
+      .then((code) =>{
+        if (code == 0) {
+          console.log('Run "gclient sync": SUCCESS');
+        } else {
+          console.log('Run "gclient sync": FAIL');
+          process.exit(1);
+        }
+      });
+};
+
+const buildTest = async () => {
+  // gn gen
+  await childProcessSpawn(
+    os.platform() == 'win32' ? 'gn.bat' : 'gn',
+    ['gen', outDir, `--args="is_debug=false webnn_enable_${backend}=true"`],
+    rootDir)
+      .then((code) =>{
+        if (code == 0) {
+          console.log(
+              `Run "gn gen ${outDir} --args=` +
+              `"is_debug=false webnn_enable_${backend}=true"": SUCCESS`);
+        } else {
+          console.log(
+              `Run "gn gen ${outDir} --args=` +
+              `"is_debug=false webnn_enable_${backend}=true"": FAIL`);
+          process.exit(1);
+        }
+      });
+
+  // ninja
+  await childProcessSpawn('ninja', ['-C', outDir], rootDir)
+      .then((code) =>{
+        if (code == 0) {
+          console.log(
+              `Run "ninja -C ${outDir}": SUCCESS`);
+        } else {
+          console.log(
+              `Run "ninja -C ${outDir}": FAIL`);
+          process.exit(1);
+        }
+      });
+
+  const outNodeDir = path.join(outDir, 'node');
+  fs.mkdirpSync(outNodeDir);
+  fs.copySync(path.join(rootDir, 'node'), outNodeDir);
+
+  process.chdir(outDir);
+
+  // Run Unit Tests
+  execFile(
+    os.platform() == 'win32' ? 'webnn_unittests.exe' : './webnn_unittests',
+    (err, data) => {
+      if (err) {
+        console.log(`Run Unit Tests: FAIL\n${err}\n${data}`);
+      } else {
+        console.log(`Run Unit Tests: SUCCESS\n${data}`);
+      }
+    });
+
+  if (backend != 'null') {
+    // Run End2End Tests
+    execFile('webnn_end2end_tests.exe', (err, data) => {
+      if (err) {
+        console.log(`Run End2End Tests: FAIL\n${err}\n${data}`);
+      } else {
+        console.log(`Run End2End Tests: SUCCESS\n${data}`);
+      }
+    });
+  }
+};
+
+(async () => {
+  await installDepotTools();
+  await syncCode();
+  await buildTest();
+})().catch((error) => console.error(error));
```

Comment:
According to https://github.com/webmachinelearning/webnn-native/runs/3073757456?check_suite_focus=true, there are 1 test case failed
```shell
[----------] Global test environment tear-down
[==========] 172 tests from 25 test suites ran. (87337 ms total)
[  PASSED  ] 171 tests.
[  FAILED  ] 1 test, listed below:
[  FAILED  ] ReduceMeanTests.ReduceMeanDefault

 1 FAILED TEST
  YOU HAVE 3 DISABLED TESTS
```

However, the build and test still passed. I think it should report error for the failures. What do you think?

Reply:
> However, the build and test still passed. I think it should report error for the failures. What do you think?

Do you mean that CI should fail on this case?
And I think CI should pass with build latest code successfully and none regression test which I'm working to implement.

Reply:
> And I think CI should pass with build latest code successfully and none regression test which I'm working to implement.

It makes sense. Let's review once you have PR for regression check.

However, I think we should have tracking issue for each failed test case. What do you think?

Reply:
> However, I think we should have tracking issue for each failed test case. What do you think?

Do you mean tracking issue on github issues? The failed ReduceMeanTests.ReduceMeanDefault test was reported https://github.com/webmachinelearning/webnn-native/issues/19.

Reply:
That's a good one. Thanks for the pointer.